neptune/internal/storage/storage_utils.py

```diff
@@ -95,8 +101,22 @@ def __repr__(self):
 
 def split_upload_files(upload_entries, max_package_size=1 * 1024 * 1024, max_files=500):
     current_package = UploadPackage()
+    walked_entries = set()
     for entry in upload_entries:
-        size = 0 if os.path.isdir(entry.source_path) else os.path.getsize(entry.source_path)
+        if os.path.isdir(entry.source_path):
```

Comment:
given a list of directories, we'll upload it's contents 1 level deep with this code.
Questions:
1. why don't we upload entire recursive dir?
2. if this is a desired behaviour for a use-case, this logic should be in the method that calls split_upload_files - it should pass a list of files without directories

Reply:
1. `os.walk` is recursive.
2. I agree, that this method coult now be split into 2 for readability.